product_deletion_utility/components/delete.py

```diff
@@ -0,0 +1,711 @@
+        is not in use by another product.
+        Args:
+            chart_name   (str): The name of the helm chart to be deleted.
+            chart_version(str): The verison of the helm chart to be removed.
+            component_nexus_id(str): Nexus ID of the helm chart to be removed.
+            nexus_api (NexusApi): The nexusctl Nexus API to interface with Nexus.
+        Returns:
+            None
+        Raises:
+            ProductInstallException: If an error occurred removing the helm chart.
+        """
+        helm_chart_short_name: str = f"{chart_name}:{chart_version}"
+        try:
+            nexus_api.components.delete(component_nexus_id)
+        except HTTPError as err:
+            if err.code == 404:
+                print(
+                    f"Helm chart {helm_chart_short_name} has already been removed")
+            else:
+                raise ProductInstallException(
+                    f"Failed to remove helm chart {helm_chart_short_name} from nexus")
+
+
+    def uninstall_loftsman_manifests(self, manifest_keys):
+        """Removes loftsman manifests for a product version from the repo.
+        Args:
+            manifests (list): List of yaml files containing loftsman manifest
+                              for deletion.
+        Returns:
+            None
+        Raises:
+            ProductInstallException: If an error occurred removing the artifact.
+        """
+        try:
+            for manifest_key in manifest_keys:
+                manifest_key = manifest_key.replace('config-data/','')
+                print(f'Removing the following manifest - {manifest_key}')
+                output = subprocess.check_output(
+                    ["cray", "artifacts", "delete", "config-data", "%s" % manifest_key], stderr=subprocess.STDOUT, universal_newlines=True)
+                print(f'Output of cray artifacts delete is {output}')
+        except subprocess.CalledProcessError as err:
+            if 'not found' in err.output:
+                print(f'Manifest {manifest_key} not available in S3 bucket config-data')
+                print(f'Output of cray artifacts delete is', err.output)
+            raise ProductInstallException(
+                f'Failed to remove loftsman manifest {manifest_key} from S3 with error: {err}'
+            )
+
+    def uninstall_ims_recipes(self, recipe_name, recipe_id):
+        """Removes ims recipes for a product version from the S3.
+        Args:
+            recipes (list of dictionaries): List of recipe names and ids
+        Returns:
+            None
+        Raises:
+            ProductInstallException: If an error occurred removing the IMS recipe.
+        """
+        try:
+            print(f'Recipe ID and name from CPC are {recipe_id}, {recipe_name}')
+            command = "cray artifacts list ims --format json | jq -r '.artifacts[] | select(.Key|test(\"^.*{}.*$\")) | .Key'".format(recipe_id)
```

Comment:
Actually we are of the understanding that it would need deletion from S3 and IMS separately. Now with your message, we got to know that these would be placed in deleted section if we only run 'cray ims recipe delete <>'. 

But as per what you have mentioned, and we also confirmed that our approach would not place things in deleted section and removes everything. This might be more efficient way for now, will keep it this way. Let us know if you foresee any issues.
![image](https://github.com/Cray-HPE/product-deletion-utility/assets/131810845/dc095d9f-b69e-46d7-8e78-0d26e641ee1c)